sdk/maps/azure-maps-render/src/test/java/com/azure/maps/render/MapsRenderClientTestBase.java

```diff
@@ -52,60 +61,67 @@ protected void beforeTest() {
 
     MapsRenderClientBuilder getRenderAsyncClientBuilder(HttpClient httpClient, MapsRenderServiceVersion serviceVersion) {
         MapsRenderClientBuilder builder = new MapsRenderClientBuilder()
+            .pipeline(getHttpPipeline(httpClient))
             .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
             .serviceVersion(serviceVersion);
-        String endpoint = getEndpoint();
-        if (getEndpoint() != null) {
-            builder.endpoint(endpoint);
-        }
-        if (getTestMode() == TestMode.RECORD) {
-            builder.addPolicy(interceptorManager.getRecordPolicy());
-        }
-        if (getTestMode() == TestMode.PLAYBACK) {
-            builder.credential(new AzureKeyCredential(FAKE_API_KEY)).httpClient(interceptorManager.getPlaybackClient());
-        } else {
-            builder.credential((new AzureKeyCredential(
-                Configuration.getGlobalConfiguration().get("SUBSCRIPTION_KEY"))));
+
+        if (interceptorManager.isPlaybackMode()) {
+            builder.endpoint("https://localhost:8080");
         }
+
         return builder;
     }
 
     HttpPipeline getHttpPipeline(HttpClient httpClient) {
-        TokenCredential credential = null;
+        httpClient = interceptorManager.isPlaybackMode() ? interceptorManager.getPlaybackClient() : httpClient;
 
-        if (!interceptorManager.isPlaybackMode()) {
-            credential = new EnvironmentCredentialBuilder().httpClient(httpClient).build();
+        if (interceptorManager.isRecordMode()) {
+            interceptorManager.addSanitizers(
+                Collections.singletonList(
+                    new TestProxySanitizer("subscription-key", ".+", "REDACTED", TestProxySanitizerType.HEADER)));
```

Comment:
This should already be redacting, is it not?

Reply:
I don't think we have released the new version of Core Test that does this. Should I make the libraries use it even if it's using the current version from source?

Reply:
I'll change the code once the new version of Core Test is released.